addons/core/functions/server/fnc_replaceSwRadiosServer.sqf

```diff
@@ -0,0 +1,66 @@
+/*
+    Name: TFAR_fnc_replaceSwRadiosServer
+
+    Author(s):
+        gandjustas
+
+    Description:
+        Handles (on server) getting switching radios, handles whether a manpack must be added to the player or not.
+
+    Parameters:
+        0: OBJECT - unit to replace radios
+
+    Returns:
+        Nothing
+
+    Example:
+        _unit remoteExec ["TFAR_fnc_replaceSwRadiosServer", 2];
+*/
+params ["_unit", "_defaultRadio"];
+private _replacedRadios = [];
+
+private _replaceRadio = {
+	params ["_radio"];
+	if (_radio == "ItemRadio") then {
+		_radio = _defaultRadio;
+	};
+
+	_radio call TFAR_fnc_instanciateRadio
+};
+
+private _fetchItems = {
+    private _allItems = ((getItemCargo (uniformContainer _this)) select 0);
+    _allItems append ((getItemCargo (vestContainer _this)) select 0);
+    _allItems append ((getItemCargo (backpackContainer _this)) select 0);
+    _allItems = _allItems arrayIntersect _allItems;//Remove duplicates
+
+    private _result = [];
+    {
+        if (_x call TFAR_fnc_isPrototypeRadio) then {
+            _result pushBack _x;
+        };
+        true;
+    } count _allItems;
+    _result
+};
+
+{
+    if (_x call TFAR_fnc_isPrototypeRadio) then {
+		//replace		
+		private _r = (_x call _replaceRadio);
+		_unit linkItem _r;
+		_replacedRadios pushBack _r;
+
+    };	
+} forEach (assignedItems _unit);
+
+{
+    if (_x call TFAR_fnc_isPrototypeRadio) then {
+		_unit removeItem _x;					
+		private _r = (_x call _replaceRadio);
+		_unit addItem _r;
```

Comment:
Uniform sync bug can randomly break his if done on any machine where the unit is not local.
This is highly unreliable.

Reply:
The impact of this Bug is almost zero with the current TFAR design.
I have no Idea if BIS will fix it, but I don't think so.

Reply:
As recommended by BIS https://community.bistudio.com/wiki/Code_Optimisation#Variable_Names I use short variable names for variables in small loops where initialization\assignment and usage are close.

Reply:
This code works for at least 2300 games and no one reports uniform bug caused by radio. No player  loose uniform on start or on take radio. For respawn - don't know, we play without respawn.

Reply:
The uniform bug is not related to TFAR.
But the bug has been around arma 3 for a long time.

Reply:
1.0 is mainly following the ace coding guidelines:
https://ace3mod.com/wiki/development/coding-guidelines.html#112-private-variable-naming

Reply:
That wiki page is not "recommended by BIS" It's nerds writing up partly idiotic and completly dumb performance improvements. Unreadibly small variable names being one of them.